src/MixedQuoter.sol

```diff
@@ -0,0 +1,325 @@
+    ) {
+        if (
+            _factoryV3 == ZERO_ADDRESS || _factoryV2 == ZERO_ADDRESS || _factoryStable == ZERO_ADDRESS
+                || _WETH9 == ZERO_ADDRESS || address(_clQuoter) == ZERO_ADDRESS || address(_binQuoter) == ZERO_ADDRESS
+        ) {
+            revert INVALID_ADDRESS();
+        }
+        factoryV3 = _factoryV3;
+        WETH9 = _WETH9;
+        factoryV2 = _factoryV2;
+        factoryStable = _factoryStable;
+        clQuoter = _clQuoter;
+        binQuoter = _binQuoter;
+    }
+
+    /**
+     * V3 *************************************************
+     */
+
+    /// @inheritdoc IPancakeV3SwapCallback
+    function pancakeV3SwapCallback(int256 amount0Delta, int256 amount1Delta, bytes memory data)
+        external
+        view
+        override
+    {
+        require(amount0Delta > 0 || amount1Delta > 0); // swaps entirely within 0-liquidity regions are not supported
+        (address tokenIn, address tokenOut, uint24 fee) = abi.decode(data, (address, address, uint24));
+        V3SmartRouterHelper.verifyCallback(factoryV3, tokenIn, tokenOut, fee);
+
+        (bool isExactInput, uint256 amountReceived) = amount0Delta > 0
+            ? (tokenIn < tokenOut, uint256(-amount1Delta))
+            : (tokenOut < tokenIn, uint256(-amount0Delta));
+
+        IPancakeV3Pool pool = V3SmartRouterHelper.getPool(factoryV3, tokenIn, tokenOut, fee);
+        (uint160 v3SqrtPriceX96After, int24 tickAfter,,,,,) = pool.slot0();
+
+        if (isExactInput) {
+            assembly {
+                let ptr := mload(0x40)
+                mstore(ptr, amountReceived)
+                mstore(add(ptr, 0x20), v3SqrtPriceX96After)
+                mstore(add(ptr, 0x40), tickAfter)
+                revert(ptr, 0x60)
+            }
+        } else {
+            /// since we don't support exactOutput, revert here
+            revert("Exact output quote not supported");
+        }
+    }
+
+    /// @dev Parses a revert reason that should contain the numeric quote
+    function parseRevertReason(bytes memory reason)
+        private
+        pure
+        returns (uint256 amount, uint160 sqrtPriceX96After, int24 tickAfter)
+    {
+        if (reason.length != 0x60) {
+            if (reason.length < 0x44) revert("Unexpected error");
+            assembly {
+                reason := add(reason, 0x04)
+            }
+            revert(abi.decode(reason, (string)));
+        }
+        return abi.decode(reason, (uint256, uint160, int24));
+    }
+
+    function handleV3Revert(bytes memory reason, IPancakeV3Pool pool, uint256 gasEstimate)
+        private
+        view
+        returns (uint256 amount, uint160 sqrtPriceX96After, uint32 initializedTicksCrossed, uint256)
+    {
+        int24 tickBefore;
+        int24 tickAfter;
+        (, tickBefore,,,,,) = pool.slot0();
+        (amount, sqrtPriceX96After, tickAfter) = parseRevertReason(reason);
+
+        initializedTicksCrossed = pool.countInitializedTicksCrossed(tickBefore, tickAfter);
```

Comment:
i think we do not need this,  
If guys agree with this , i will remove related codes

Reply:
can you create an issue around this if you feel it should be removed? 

though lets keep it in this PR to make it easier to pass the review and approve (eg. burger/mist verify the code is copied from existing from MixedQuoterV1 + add v4 logic)

Reply:
also can i trouble you to check with Jackson if FE uses this? I see you are thinking of removing this because `quoteMixedExactInput` does not return that anymore..

If FE might use it, i might prefer to add this in `quoteMixedExactInput`

Reply:
ok , let us keep this.